### PR TITLE
Adjustments to Azure TRE docs

### PR DESCRIPTION
**Pull request by anastasiakuzn**
##The following changes to AzureTRE Docs are being addressed:

These Azure TRE Adjustments include the following:

**Navigation**

- Hide the Compliance sub-section from the navigation pane from Overview section
- Change the name of Using the TRE for Research section to Using the Azure TRE
- Move the following sub-sections from Using the TRE for Research to Technical Guide:
    - Local Development: using-tre/local-development/index.md
    - CI/CD pipelines: using-tre/pipelines/index.md

**Content**

- Add Airlock Manager role to User Roles sub-section

### CODE REVIEW

File: mkdocs.yml [hunk 1]
```diff
@@ -135,6 +133,9 @@ nav:
 
           - Releases: tre-developers/release.md
 
+      - Local Development: using-tre/local-development/index.md
```
Comment (marrobi) [inline mkdocs.yml:136]: I think Local Development needs to go under the `Development` heading.

There is also a CI/CD link here:

image

Makes sense to have a single CI/CD section under Technical Guide - CI/CD isn't just for development, but deployment.

**Review by marrobi — COMMENTED** [review]
Couple of comments.

File: docs/index.md [hunk 2]
```diff
@@ -1,5 +1,10 @@
 # Azure TRE Overview
 
+## What is Azure TRE?
+
+Azure Trusted Research Environment refers to a secure research environment intended to allow researchers to access sensitive data under a higher level of control and data protection. The main objective of Azure TRE is to proivde a computing environment with controlled access to data for projects that require high security and the ability to audit.
```
[the remark this answers is not in the document]
Comment (anastasiakuzn) [reply docs/index.md]: @marrobi I have removed the index.md file from this PR for now for further iteration as we discussed.

File: mkdocs.yml (hunk 1 above)
Comment (anastasiakuzn) [reply mkdocs.yml]: @marrobi - moved the "Local Development" under "Technical Guide" section and "Development" sub-section.

FO CI/CD - would you like me to create a new "sub-section" under "Technical Guide" which would include the whole sub-section of "Deploying the AzureTRE using CI/CD" and "CI / CD Pipelines" ?

Comment (marrobi) [reply mkdocs.yml]: Are they not both the same thing?

Comment (anastasiakuzn) [reply mkdocs.yml]: @marrobi, after having another read through CI/CD documentation my suggestion is to move "CI/CD Pipelines" sub-section under "Deploying the AzureTRE using CI/CD".
image

I think it is an appropriate place for this sub-section "Deploying the AzureTRE using CI/CD" does not have any content except for "Build Runners" sub-section.

**Review by marrobi — APPROVED** [review]
Thanks, makes more sense.

**Pull request merged**